src/components/icon/icon.tsx

```diff
@@ -1,5 +1,21 @@
-import type { FC } from "react";
+import { forwardRef, type SVGAttributes } from "react";
+import { type IconKeys, Icons } from "./resolver.js";
 
-export const Icon: FC = () => {
-  return <span role="img">icon</span>;
-};
+export interface IconProps extends Omit<SVGAttributes<SVGElement>, "name"> {
+  name: IconKeys;
+  svgName?: string;
+}
+
+export const Icon = forwardRef<SVGSVGElement, IconProps>((props, ref) => {
+  const { name: iconName, svgName: name, ...rest } = props;
```

Comment:
why these prop name reassignments? what does svgName do?

Reply:
svg has a prop name by default

Reply:
but why do you rename these props internally? just use them as they are:

```tsx
const Component = Icons[name];
...
<Component ref={ref} name={svgName} {...props} />;
```

consider destructuring within the function args directly instead of a separate line. and ...props instead of ...rest:

```tsx
export const Icon = forwardRef<SVGSVGElement, IconProps>(({name, svgName, ...props}, ref) => {
```